shotover-proxy/src/transforms/redis/cache.rs

```diff
@@ -1,23 +1,81 @@
 use crate::config::topology::TopicHolder;
 use crate::error::ChainResponse;
-use crate::frame::{CassandraFrame, CassandraOperation, CassandraResult, Frame, RedisFrame};
-use crate::message::{Message, MessageValue, Messages, QueryType};
+use crate::frame::cassandra::CQLStatement;
+use crate::frame::{CassandraFrame, CassandraOperation, CassandraResult, Frame, RedisFrame, CQL};
+use crate::message::{Message, Messages, QueryType};
 use crate::transforms::chain::TransformChain;
 use crate::transforms::{
     build_chain_from_config, Transform, Transforms, TransformsConfig, Wrapper,
 };
-use anyhow::{anyhow, bail, Result};
+use anyhow::Result;
 use async_trait::async_trait;
-use bytes::{BufMut, Bytes, BytesMut};
+use bytes::{Bytes, BytesMut};
+use cassandra_protocol::frame::Serialize;
 use cassandra_protocol::frame::Version;
+use cql3_parser::cassandra_statement::CassandraStatement;
+use cql3_parser::common::{Operand, RelationElement, RelationOperator};
+use cql3_parser::select::{Named, Select, SelectElement};
 use itertools::Itertools;
+use metrics::{register_counter, Counter};
 use serde::Deserialize;
-use sqlparser::ast::{Assignment, BinaryOperator, Expr, Ident, Query, SetExpr, Statement, Value};
-use std::borrow::Borrow;
-use std::collections::HashMap;
+use std::collections::{BTreeMap, HashMap};
+use std::fmt::{Display, Formatter};
+use std::io::Cursor;
+use std::ops::Deref;
+use tracing_log::log::{debug, error, info, trace, warn};
+
+/*
+Uses redis as a cache.  Data is stored in Redis as a Hash.
+The key for the Redis cache is the key for the Cassandra query.
+Only exact Redis matches are supported, though the redis key can equate to a Cassandra scan.
+If the key is requested again the result query result is returned from the cache.
+If the key is deleted or updated the key is removed from the cache.
+If the table is dropped the keys are removed from the cache.
+
+The redis hash keys are:
+data - serialized form of the row data from cassandra.
+metadata - serialized form of the metadata from cassandra.
+
+ */
+enum CacheableState {
+    /// string is the table name
+    Read(String),
```

Comment:
When its not obvious what the inner value means it makes sense to rewrite as a field:
e.g. `Read { table_name: String }`

For a variant like `Err(String)` its fine as is because its obvious the value will contain an error message.

Reply:
Yeah, you would want a `Skip { reason: String }` because thats not obvious too.
And at that point you may as well have Err as `Err { reason: String }` for consistency.